### PR TITLE
GPII-3103 README improvements.

### PR DESCRIPTION
**Pull request by mrtyler**
This work was motivated by https://issues.gpii.net/browse/GPII-3103 but includes a number of cleanups based on Cindy's feedback while working with gpii-infra, plus some other edits and updates.

Reading raw markdown is no fun, so here's the rendered version:
https://github.com/mrtyler/gpii-infra/blob/GPII-3103-readme/aws/README.md

### CODE REVIEW

**Comment by mrtyler** [comment]
Heh, I can't "request a review" from @cindyli, I guess because she's not in the org?

**Comment by natarajaya** [comment]
LGTM

File: aws/README.md [hunk 1]
```diff
@@ -173,30 +173,52 @@ To delete the lock:
...
+   * Not all cloud resources are Taggable so you may need to explore a little, but the Resource Group report should give you an idea of what kinds of resources are getting stuck.
+   * You can also use the new (non-Classic) Resource Group interface, though I find it more confusing than Classic Resource Groups:
+      * Top bar `->` Resource Groups `->` Create a Resource Group.
+      * Leave "Select resource types" dropdown alone.
+      * "Tag key" is `KubernetesCluster`.
+      * "Optional tag value" is `dev-mrtyler.gpii.net`.
+      * Click "View group resources" above the filter fields.
 1. The AWS dashboard, part 2 - various tools in the system store state in S3 and DynamoDB. If you encounter weird mismatch errors, you may need to perform more manual cleanup.
-   * Check S3 Bucket `gpii-infra-state` for Keys named after your environment (`dev-mrtyler.gpii.net`) and delete only those keys. Remember to check non-environment subdirectories like `prereqs`.
+   * Check S3 Bucket `gpii-terraform-state` for Keys named after your environment (`dev-mrtyler.gpii.net`) and delete only those keys.
```
Comment (cindyli) [inline aws/README.md]: On my notes, before deleting S3 buckets, there's this step:
```
Delete records with “cindyli” in this order: Auto scaling groups -> instances -> volumes -> VPC -> Security group
* "Auto scaling groups, instances, volumes" can be found on “EC2 dashboard”
“Security group” and “VPC” can be found on: Top left menu “service” -> VPC -> left side menu “security group” -> delete all instances with “cindyli” (VPC will be auto deleted once all security groups are gone)
```

**Comment by amatas** [comment]
LGTM

**Comment by mrtyler** [comment]
I incorporated Cindy's notes (thanks Cindy!) and committed a small change I forgot in CI-CD.md.

**Comment by cindyli** [comment]
LGTM. Thanks, @mrtyler.

**Pull request merged**